include/ncengine/physics/RigidBody.h

```diff
@@ -93,7 +86,10 @@ auto SetSimulatedBodyScale(Transform& transform,
 class RigidBody
 {
     public:
-        RigidBody(Entity self, Shape shape, BodyType bodyType, RigidBodyFlags::Type flags = RigidBodyFlags::Default)
+        RigidBody(Entity self,
+                  const Shape& shape = Shape::MakeBox(),
```

Comment:
If move were meaningful to `Shape`, I'd take by value and move into the member here, but its full of primitives. Since R values and L values can bind to const &, taking by const & allows the caller to give us an temporary, but also avoids an extra copy in the case where an L value is passed in (copy once from call site to the parameter, then again from parameter to the member). If an R value is passed, we get the double copy, but we can't do anything about that. Although, I imagine the compiler can optimize that just fine. If we get the default param, its the same as if we were given an R value.